apps/web/src/features/upload/utils.ts

```diff
@@ -138,7 +142,31 @@ export function getZodTypeName(schema: z.ZodTypeAny, isOptional?: boolean): ZodT
       };
     } else if (isZodArrayDef(def)) {
       return interpretZodArray(schema, def.typeName, isOptional);
+    } else if (isZodSetDef(def)) {
+      const innerDef = def.valueType._def as AnyZodTypeDef;
```

Comment:
We should not introduce any new code like this. Please write a type safe function to validate and parse the `_def`. This can also fix other type issues that we have.

Reply:
For this feature to no longer be marked as experimental, there should be no `as` statements, except with detailed comment expressing why this is necessary.